packages/core/src/types/provider.ts

```diff
@@ -148,32 +148,46 @@ export type AssetIdOrString =
       assetId: string;
     };
 
-export type CreateAssetArgs = {
+export type CreateAssetFileProgress = {
+  /** Name of the asset */
+  name: string;
+  /** Progress from 0 to 1 */
+  progress: number;
+};
+
+export type CreateAssetProgress = {
+  /** Average of the progress values */
+  average: number;
+  /** Progress values */
+  sources: CreateAssetFileProgress[];
+};
+
+export type CreateAssetSourceBase = {
   /** Name for the new asset */
   name: string;
 
-  /** External URL to be imported */
-  url?: string;
+  /** Metadata associated with the asset */
+  meta?: Record<string, string>;
+};
+
+export type CreateAssetSourceUrl = CreateAssetSourceBase & {
+  /** URL of the asset to import */
+  url: string;
+};
 
-  /** Content to be uploaded */
-  file?: File | ReadStream;
-  /** Size of the upload file. Must provide this if the file is a ReadStream */
+export type CreateAssetSource = CreateAssetSourceBase & {
+  /** Content to be uploaded or streamed */
+  file: File | ReadStream;
+  /** Size of the file, required if this is a stream. */
   uploadSize?: number;
```

Comment:
Let's actually remove this field entirely - as of 3.0.0, it is not a required field.

https://tus.io/blog/2022/08/03/tus-js-client-300.html